velox/connectors/hive/storage_adapters/hdfs/tests/HdfsFileSystemTest.cpp

```diff
@@ -139,43 +146,57 @@ void verifyFailures(hdfsFS hdfs) {
            "(%d vs. %d) Cannot read HDFS file beyond its size: %d, offset: %d, end point: %d") %
        size % endpoint % size % startPoint % endpoint)
           .str();
-  auto serverAddress = (boost::format("%s:%s") % localhost % hdfsPort).str();
+
   auto readFailErrorMessage =
       (boost::format(
-           "Unable to open file %s. got error: HdfsIOException: InputStreamImpl: cannot open file: %s.\t"
-           "Caused by: Hdfs::HdfsRpcException: HdfsFailoverException: Failed to invoke RPC call \"getBlockLocations\" on server \"%s\"\t\t"
-           "Caused by: HdfsNetworkConnectException: Connect to \"%s\" failed") %
-       destinationPath % destinationPath % serverAddress % serverAddress)
-          .str();
-  auto builderErrorMessage =
-      (boost::format(
-           "Unable to connect to HDFS: %s, got error: Hdfs::HdfsRpcException: HdfsFailoverException: "
-           "Failed to invoke RPC call \"getFsStats\" on server \"%s\"\tCaused by: "
-           "HdfsNetworkConnectException: Connect to \"%s\" failed") %
-       serverAddress % serverAddress % serverAddress)
+           "Unable to open file %s. got error: ConnectException: Connection refused") %
+       destinationPath)
           .str();
+
   checkReadErrorMessages(&readFile, offsetErrorMessage, kOneMB);
   HdfsFileSystemTest::miniCluster->stop();
-  checkReadErrorMessages(&readFile2, readFailErrorMessage, 1);
-  try {
-    auto config = std::make_shared<const config::ConfigBase>(
-        std::unordered_map<std::string, std::string>(configurationValues));
-    filesystems::HdfsFileSystem hdfsFileSystem(
-        config,
-        filesystems::HdfsFileSystem::getServiceEndpoint(
-            simpleDestinationPath, config.get()));
-    FAIL() << "expected VeloxException";
-  } catch (VeloxException const& error) {
-    EXPECT_THAT(error.message(), testing::HasSubstr(builderErrorMessage));
+
+  const int max_retries = 10;
+  const int sleep_duration_ms = 1000;
+  int retries = 0;
+  while (true) {
+    if (retries >= max_retries) {
```

Comment:
This is similar to the previous sleep(10). I was thinking of the following
```
  // Wait until the miniCluster is shutdown to ensure consistency
  // in connection status.
  // This prevents a scenario where the first pread call in the
  // checkReadErrorMessages method might succeed, while a subsequent call fails,
  // leading to a mismatch in readFailErrorMessage.
  constexpr auto kMaxRetries = 10;
  int retries = 0;
  while (true) {
    if (!HdfsFileSystemTest::miniCluster->isRunning()) {
      checkReadErrorMessages(&readFile2, readFailErrorMessage, 1);
      break;
    } else {
      if (retries >= kMaxRetries ) {
          FAIL() << "expected VeloxException";
      } else {
          sleep(1);
          retries++;
      } 
      continue;
    }
  }
```

Reply:
@majetideepak The miniCluster->isRunning() method will return false after calling miniCluster->stop(). However, the miniCluster does not shut down immediately. So the error message mismatch issue still exists in readFailErrorMessage method.

Reply:
@JkSelf I see. Is there a way we can check when the miniCluster is shut down? Maybe using some curl command?

Reply:
@majetideepak Yes. Updated. Can you help to review again? Thanks.